pinot-core/src/main/java/com/linkedin/pinot/core/io/reader/impl/v1/BaseChunkSingleValueReader.java

```diff
@@ -105,17 +131,17 @@ protected ByteBuffer getChunkForRow(int row, ChunkReaderContext context) {
       chunkSize = nextChunkOffset - chunkPosition;
     }
 
-    ByteBuffer uncompressedBuffer = context.getChunkBuffer();
-    uncompressedBuffer.clear();
+    ByteBuffer decompressedBuffer = context.getChunkBuffer();
+    decompressedBuffer.clear();
 
     try {
-      _chunkDecompressor.decompress(_dataBuffer.toDirectByteBuffer(chunkPosition, chunkSize), uncompressedBuffer);
+      _chunkDecompressor.decompress(_dataBuffer.toDirectByteBuffer(chunkPosition, chunkSize), decompressedBuffer);
     } catch (IOException e) {
       LOGGER.error("Exception caught while decompressing data chunk", e);
```

Comment:
having chunkPosition and some reference to PinotDataBuffer will help in debugging if there is a corruption